amethyst_core/src/timing.rs

```diff
@@ -50,24 +70,58 @@ impl Time {
         self.last_fixed_update
     }
 
-    /// Gets the total number of frames that have been played in this session.
+    /// Gets the time since the start of the game, taking into account the speed multiplier.
+    pub fn absolute_time(&self) -> Duration {
+        self.absolute_time
+    }
+
+    /// Gets the time since the start of the game as seconds, taking into account the speed multiplier.
+    pub fn absolute_time_seconds(&self) -> f32 {
+        duration_to_secs(self.absolute_time)
+    }
+
+    /// Gets the time since the start of the game, ignoring the speed multiplier.
+    pub fn absolute_real_time(&self) -> Duration {
+        self.absolute_real_time
+    }
 
+    /// Gets the time since the start of the game as seconds, ignoring the speed multiplier.
+    pub fn absolute_real_time_seconds(&self) -> f32 {
+        duration_to_secs(self.absolute_real_time)
+    }
+
+    /// Gets the current time speed multiplier.
+    pub fn time_multiplier(&self) -> f32 {
+        self.time_multiplier
+    }
+
+    /// Gets the total number of frames that have been played in this session.
     /// Sets both `delta_seconds` and `delta_time` based on the seconds given.
     ///
     /// This should only be called by the engine.  Bad things might happen if you call this in
     /// your game.
     pub fn set_delta_seconds(&mut self, secs: f32) {
-        self.delta_seconds = secs;
-        self.delta_time = secs_to_duration(secs);
+        self.delta_seconds = secs * self.time_multiplier;
+        self.delta_time = secs_to_duration(secs * self.time_multiplier);
+        self.delta_real_seconds = secs;
+        self.delta_real_time = secs_to_duration(secs);
+
+        self.absolute_time += self.delta_time;
+        self.absolute_real_time += self.delta_real_time;
     }
 
     /// Sets both `delta_time` and `delta_seconds` based on the duration given.
     ///
     /// This should only be called by the engine.  Bad things might happen if you call this in
     /// your game.
     pub fn set_delta_time(&mut self, time: Duration) {
-        self.delta_seconds = duration_to_secs(time);
-        self.delta_time = time;
+        self.delta_seconds = duration_to_secs(time) * self.time_multiplier;
```

Comment:
I would prefer if we don't change the semantics of an established field in this way, please change so the old fields retain their functionality, and add the time scale to new fields instead.

Reply:
I personally think it makes more sense to have time affected by the multiplier by default. It won't affect ANY game nor the engine. To affect a game, the user needs to set the speed multiplier himself. To affect the engine... well I already fixed the places where it was using delta_something and it was supposed to use real time.
The cases where you want to have real time make more sense into the engine, as it is what "controls" the simulation. End-users won't even notice the difference :)

Reply:
@Rhuagh I'm not sure if a `scaled_delta_time` is the right thing here, as this kind of functionality can make it really easy to quickly introduce pausing functionality to a game by setting the time scale to 0.  So I'd actually prefer we encourage usage of a scaled delta time by default with the default scaling set to 1.

Unless the end user asks to scale the time, this doesn't change the use of the field.

Reply:
I can accept that.